gpflow/training/monitor.py

```diff
@@ -1050,8 +1050,15 @@ def __init__(self, file_writer: LogdirWriter, func: Callable[[], Figure], func_n
 
         super().__init__(file_writer)
         self.func = func
-        self.placeholder = tf.placeholder(tf.float64, [1, None, None, None])
-        self._summary = tf.summary.image(func_name, self.placeholder)
+
+        self.buf_placeholder = tf.placeholder(tf.string)
+        # Create TF image and load its content from the buffer.
+        image_tensor = tf.image.decode_png(self.buf_placeholder)
```

Comment:
```suggestion
        self.image_tensor = tf.image.decode_png(self.buf_placeholder)[None]
```

Reply:
Could potentially cleanup the code a bit. If you accepts this change, you can delete the next two lines.

Reply:
Cleaned it up even more, feeding the image_tensor directly to tf.summary.image() instead of having an intermediate tf.placeholder and two separate session.run() calls